### DISCUSSION TITLE
Delete/cleanup the whole history of an object

### DISCUSSION THREAD

**Discussion by Asky-GH** [javers/javers#1126] [Q&A] [answered]
I see four methods related to **delete** functionality in Javers interface. But according to javadocs, if I'm not mistaken, these methods don't delete any records from the database. Is there a possibility to clean up history data of objects that I'm not interested in anymore?

**Comment by bartoszwalacik** [comment, marked as answer]
You are right, Javers' repository is the append only collections of snapshots. Marking an object  as deleted creates a terminal snapshot for it. There is an open feature request for implementing `javers.deleteAllSnapshots(Class entity)` see https://github.com/javers/javers/issues/1107

**Reply by Asky-GH** [reply]
@bartoszwalacik I've noticed that there is a commit that references this open feature request. And it looks like it's been a while since the commit was added. I wonder why it is not being merged yet. Is it because of some issues in the implementation or maybe the problem is in the absence of tests?

**Reply by bartoszwalacik** [reply]
which one?

**Reply by Asky-GH** [reply]
https://github.com/evfool/javers/commit/706428a7de4c05b0a3caa1c00b0f24c622b1e014

**Reply by bartoszwalacik** [reply]
@Asky-GH this commit looks like a sketch, I dont think it will work, and there are no tests
